specs/src/tx_client_v2.md

```diff
@@ -0,0 +1,148 @@
+  - context
+  - a result channel
+- The queue enforces memory based backpressure:
+  - If sufficient memory is available, the job is appended to the queue
+  - If queue exceeds its memory limit, submission blocks until memory is freed up
+ User should be able to set desired queue size during client initialization.
+- Nothing gets signed yet.
+- No network operations are performed.
+- User will receive tx result asynchronously via the result channel.
+
+2. Selecting the next transaction to broadcast
+
+The submitter processes transactions strictly in submission order:
+
+- Submission proceeds only if the queue is not in recovery mode.
+- The next unbroadcast transaction is selected.
+- The transaction is built and signed using the next sequence number.
+- The transaction is broadcast to the node.
+
+If broadcast succeeds:
+
+- Tx hash, sequence, and signed bytes are recorded directly on the queued tx.
+- Tx is now considered pending.
+
+If broadcast fails due to a sequence mismatch:
+
+- Submitter pauses further submissions.
+- Job remains in the queue.
+- Waits for confirmation/recovery logic to unblock submissions.
+
+Sequence mismatches during broadcast are treated as coordination signals not terminal errors. If client remains blocked for longer than a minute, then it's possible to parse expected sequence from the error message and resign as recovery mech.
+
+If broadcast fails for any other reason:
+
+- Tx is removed from the queue.
+- An error is returned to the user via the result channel.
+
+### Phase 2: Confirmation
+
+Confirmation runs asynchronously at a configurable polling interval.
+
+The monitor periodically queries the chain for the status of broadcast transactions, performing status checks in bounded batches starting from the first pending transaction.
+
+There are four distinct statuses:
+
+1. Pending
+
+- No action needed
+- Move to the next
+
+1. Committed
+
+- The tx was included in a block.
+- The user receives a `TxResponse` populated with execution and metadata returned by the chain (e.g. height, hash, execution code, gas usage, and signers).
+- The tx is removed from the queue.
+
+If execution failed:
+
+- The tx is removed from the queue.
+- The user receives a execution error containing the error code and log.
+
+1. Evicted
+
+- Queue enters recovery mode, preventing new transactions from being broadcast.
+- Evicted tx is marked as being resubmitted to avoid duplicate attempts.
+- Tx is sent to an internal resubmission channel, which is handled by the queue's coordinator that also handles submissions.
+
+Resubmission behavior:
+
+- Tx is re-broadcast using the same signed tx bytes.
+- Tx is **never** resigned.
+- Resubmitted tx remains as pending in queue.
+- Recovery mode is exited.
+- Broadcasting unblocks.
+
+1. Rejected
+
+Sequence recovery:
+A rejected transaction means the node will be expecting a lower sequence for this account. To unblock submissions the client must update its local sequence to match the node.
```

Comment:
> A rejected transaction means the node will be expecting a lower sequence for this account. 

is this true? my understanding is that the new strategy is simply resubmit, which we should be able to do due to the queue

Reply:
Yes it is true. We can still have rejected transactions in ReCheck e.g user ran out of funds.